### ISSUE TITLE
Auto adjust text to fit Text node width

### ISSUE THREAD

**Issue by tomkur** [facebook/react-native#5877]
This is similar to #728, but instead of adjusting font size, I'd like to adjust the text itself. E.g. imagine you're presenting some data for 6 selected cities and you want to show names of these cities to the user, but have only enough space to fit 2 of them. So you might want to show:
`Atlanta, Chicago, + 4 more`
instead of
`Atlanta, Chicago, Los Ange`
or
`Atlanta, Chicago, Los A...`

I've created [this gist](https://gist.github.com/tomkur/0446549bf5177433bb33), which is based on [this one](https://github.com/facebook/react-native/issues/728#issuecomment-123126144) created by @brentvatne's (big thanks for that example!) and it seems to solve this problem, but it would be good if someone checked if this is going in the appropriate direction, so I could create a PR.

**Comment by tomkur** [comment]
Screenshot from that gist run on RN 0.19, iPhone 6 simulator:
<img width="381" alt="screen shot 2016-02-11 at 15 13 31" src="https://cloud.githubusercontent.com/assets/2394102/12978599/2c3a10cc-d0d2-11e5-9896-616330c8ca1b.png">

**Comment by skevy** [comment]
@tomkur there's already a PR open for this #2327. Could you comment on it/on the issue (#728) if you have thoughts on how to push it forward?

Closing in favor of #728.

**Comment by tomkur** [comment]
As far as I understand, #4026 is about resizing font to fit, whereas this one is about shortening text to fit. Do you think I should comment there anyway?

**Comment by MattFoley** [comment]
If possible, I'd love to merge something like this with my current PR on the auto sizing to implement ... truncation. I need a bit of help, haven't had time to finish it on my own. I think the "+ X items" is a good idea, but might be used less often than truncation.

**Comment by stoffern** [comment]
Created a solution for this here: https://github.com/stoffern/react-native-textfit
cc @tomkur @MattFoley

**Comment by MattFoley** [comment]
@stoffern Thanks for the heads up, note that I've suggested that https://github.com/facebook/react-native/pull/4026 is ready to merge, and am currently just waiting on a reviewing to tackle it. I've been using the solution in #4026 for 6-8mo and don't plan on changing our implementation unless we decide to drop <Text>, and just create our own UILabel wrapper.

Originally the change was made natively because it should be much more performant, but the fact that your solution is cross platform is nice.
